src/utils/download.ts

```diff
@@ -0,0 +1,25 @@
+import * as fs from 'fs';
+import path from 'path';
+import axios, { AxiosResponse } from 'axios';
+
+/**
+ * Downloads file from remote HTTP[S] host and puts its contents to the
+ * specified location.
+ * @param url URL of the file to download.
+ * @param destinationPath Path to the destination file.
+ */
+export async function download(url: string, filePath: string): Promise<string> {
+	const fileName = url.split('/').pop();
+	if (!fileName) {
+		throw new Error('Invalid remote path. No file name found.');
```

Comment:
Why is obtaining a file name a requirement for this function? What if someone just wants to download the index page that gets returned from "https://whatever.com"? That seems valid.

I would have thought we would just generate a temporary filename in a tmp folder somewhere, and then have axios stream to that spot. Assuming that doesn't throw, then we use that tmp local filepath (whose name can be absolutely random... who cares?) to feed our normal upload code with whatever naming rules we want to apply THERE, rather than here. This should just be a utility downloader function. Filename derivation logic need not apply here.